WeightProducer/python/MCSampleValues.py

```diff
@@ -63,493 +63,1039 @@ class MCSampleValuesHelper():
+                kFactor_2017=1.1374, kFactorSource_2017="XSDB NNLO/LO=6077.22/5343",
+                kFactor_2018=1.1421, kFactorSource_2018="XSDB NNLO/LO=6077.22/5321",
+            ),
         },
         "DYJetsToLL_M-50_HT-800to1200" : {
-            "CrossSection" : XSValues(XS_13TeV=0.775392),
-            "kFactor" : kFactorValues(kFactor_13TeV=1.23),
+            "CrossSection" : XSValues(
+                XS_2016=0.6229, XSSource_2016="XSDB (LO)",
+                XS_2017=0.8052, XSSource_2017="XSDB (LO)",
+                XS_2018=0.8094, XSSource_2018="GenXSecAnalyzer",
+            ),
+            "kFactor" : kFactorValues(
+                kFactor_2016=1.2245, kFactorSource_2016="XSDB NNLO/LO=6077.22/4963",
+                kFactor_2017=1.1374, kFactorSource_2017="XSDB NNLO/LO=6077.22/5343",
+                kFactor_2018=1.1421, kFactorSource_2018="XSDB NNLO/LO=6077.22/5321",
+            ),
         },
         "DYJetsToLL_M-50_HT-1200to2500" : {
-            "CrossSection" : XSValues(XS_13TeV=0.186222),
-            "kFactor" : kFactorValues(kFactor_13TeV=1.23),
+            "CrossSection" : XSValues(
+                XS_2016=0.1512, XSSource_2016="XSDB (LO)",
+                XS_2017=0.1933, XSSource_2017="XSDB (LO)",
+                XS_2018=0.1931, XSSource_2018="GenXSecAnalyzer",
+            ),
+            "kFactor" : kFactorValues(
+                kFactor_2016=1.2245, kFactorSource_2016="XSDB NNLO/LO=6077.22/4963",
+                kFactor_2017=1.1374, kFactorSource_2017="XSDB NNLO/LO=6077.22/5343",
+                kFactor_2018=1.1421, kFactorSource_2018="XSDB NNLO/LO=6077.22/5321",
+            ),
         },
         "DYJetsToLL_M-50_HT-2500toInf" : {
-            "CrossSection" : XSValues(XS_13TeV=0.00438495),
-            "kFactor" : kFactorValues(kFactor_13TeV=1.23),
+            "CrossSection" : XSValues(
+                XS_2016=0.003659, XSSource_2016="XSDB (LO)",
+                XS_2017=0.003468, XSSource_2017="XSDB (LO)",
+                XS_2018=0.003514, XSSource_2018="GenXSecAnalyzer",
+            ),
+            "kFactor" : kFactorValues(
+                kFactor_2016=1.2245, kFactorSource_2016="XSDB NNLO/LO=6077.22/4963",
+                kFactor_2017=1.1374, kFactorSource_2017="XSDB NNLO/LO=6077.22/5343",
+                kFactor_2018=1.1421, kFactorSource_2018="XSDB NNLO/LO=6077.22/5321",
+            ),
         },
         "DYJetsToLL_M-50" : {
-            "CrossSection" : XSValues(XS_13TeV=6025.2),
+            "CrossSection" : XSValues(
+                XS_13TeV=6077.22, XSSource_13TeV="XSDB (NNLO)",
+            ),
         },
         "ZJetsToNuNu_HT-100To200" : {
-            "CrossSection" : XSValues(XS_13TeV=344.8305),
-            "kFactor" : kFactorValues(kFactor_13TeV=1.23),
+            "CrossSection" : XSValues(
+                XS_2016=93.35, XSSource_2016="XSDB (LO)",
+                XS_2017=302.8, XSSource_2017="XSDB (LO)",
```

Comment:
really a 3x jump 2016 -> 2017?

Reply:
This has to do with the 2016 sample being generated using only a single lepton flavor and the 2017/2018 samples using all three lepton flavors. The 2016 cross section values will be multiplied by 3. For reference, I've put the Madgraph process fragments below:

2016:
```
generate p p > ve ve~ / h @0
add process p p > ve ve~ j / h @1
add process p p > ve ve~ j j / h @2
add process p p > ve ve~ j j j / h @3
add process p p > ve ve~ j j j j / h @4
```

2017/2018:
```
generate    p p > vl vl~ / h @0
add process p p > vl vl~ j / h @1
add process p p > vl vl~ j j / h @2
add process p p > vl vl~ j j j / h @3
add process p p > vl vl~ j j j j / h @4
```